internal/controllers/topology/cluster/patches/variables/variables.go

```diff
@@ -64,6 +67,20 @@ type ClusterTopologyBuiltins struct {
 	Class string `json:"class,omitempty"`
 }
 
+// ClusterNetworkBuiltins represents builtin cluster network variables.
+type ClusterNetworkBuiltins struct {
+	// APIServerPort is the port the API Server should bind to.
+	APIServerPort *int32 `json:"apiServerPort,omitempty"`
```

Comment:
@fabriziopandini Given the ongoing discussion around APIServer local/lb ports. Do you think it's safe to provide this as a builtin variable? (or another point of view, if we don't have demand for it, maybe we should omit it given that we might want to change the API soon)

Reply:
@yastij opinions?
I don't see this going away soon, but at the same time holding back people from using it in patch makes sense if the plan is to deprecate it

Reply:
I'd rather not expose it as a patch and have the ControlPlaneEndpoint + the new `LocalAPIServerPort` exposed. at least folks moving to clusterClass would be using the correct fields for port management

Reply:
I think we can leave it out of this PR and if it there's demand for it we can decide whether or not to add it in later.